### PR TITLE
Order by finder

### PR DESCRIPTION
**Pull request by ikwattro**
Added the possibility to add order by to the queries via the `findBy` method : 

```php
$users = $em->getRepository(User::class)->findBy([], ['name' => 'ASC']);
```

### CODE REVIEW

File: src/Persisters/BasicEntityPersister.php [hunk 1]
```diff
@@ -150,6 +150,15 @@ public function getMatchCypher(array $criteria = [], $limit = null, $offset = nu
 
         $cypher .= 'RETURN '.$identifier;
 
+        if (is_array($orderBy) && count($orderBy) > 0) {
+            $cypher .= PHP_EOL;
+            $i = 0;
+            foreach ($orderBy as $property => $order) {
+                $cypher .= $i === 0 ? 'ORDER BY ' : ', ';
```
Comment (Nyholm) [inline src/Persisters/BasicEntityPersister.php]: Should be `$cypher .= $i++ === 0 `

Or add `$i++` on a new row for readability.

File: src/Persisters/BasicEntityPersister.php [hunk 2]
```diff
@@ -132,7 +132,7 @@ public function getRelationshipEntityCollection($alias, $sourceEntity)
      *
      * @return Statement
      */
-    public function getMatchCypher(array $criteria = [], $limit = null, $offset = null, $orderBy = null)
+    public function getMatchCypher(array $criteria = [], $orderBy = null, $limit = null, $offset = null)
```
Comment (Nyholm) [inline src/Persisters/BasicEntityPersister.php]: Make sure to update doc block.

File: src/Persisters/BasicEntityPersister.php (hunk 1 above)
Comment (ikwattro) [inline src/Persisters/BasicEntityPersister.php]: good point

Comment (ikwattro) [inline src/Persisters/BasicEntityPersister.php]: addressed

File: src/Persisters/BasicEntityPersister.php (hunk 2 above)
Comment (ikwattro) [inline src/Persisters/BasicEntityPersister.php]: adressed

Comment (ikwattro) [inline src/Persisters/BasicEntityPersister.php]: @Nyholm I added in load as well.

**Pull request merged**